dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,459 @@
+        """ Triggers a copy operation for all restored objects passed in @id list
+
+        :param atid_list: list of @ids or actual file metadata objects to issue copy operations back to their
+                          original bucket/key location
+        :param storage_class: which storage class to copy into
+        :param parallel: whether to parallelize the copy
+        :param num_threads: number of threads to use when parallelizing, default to 4
+        :return: 2 tuple of success, error list of @ids
+        """
+        success, errors = [], []
+        if parallel:
+            with ThreadPoolExecutor(max_workers=num_threads) as executor:
+                futures = []
+                for atid in atid_list:
+                    if isinstance(atid, dict):
+                        _atid = atid['@id']
+                    else:
+                        _atid = atid
+                    files_meta = self.resolve_bucket_key_from_portal(_atid, atid)
+                    for bucket, key in files_meta:
+                        future = executor.submit(self.copy_object_back_to_original_location, bucket, key, storage_class)
+                        futures.append(future)
+                for future in tqdm(futures, total=len(atid_list)):
+                    res = future.result()
+                    if res:
+                        success.append(res)
+                    else:
+                        errors.append(res)
+        else:
+            for atid in atid_list:
+                if isinstance(atid, dict):
+                    _atid = atid['@id']
+                else:
+                    _atid = atid
+                files_meta = self.resolve_bucket_key_from_portal(_atid, atid)
+                accumulated_results = []
+                for bucket, key in files_meta:
+                    resp = self.copy_object_back_to_original_location(bucket, key, storage_class)
+                    if resp:
+                        accumulated_results.append(_atid)
+                if len(accumulated_results) == len(files_meta):  # all files for this @id were successful
+                    success.append(_atid)
+                else:
+                    errors.append(_atid)
+        if len(errors) != 0:
+            PRINT(f'Errors encountered copying @ids: {errors}')
+        else:
+            PRINT(f'Successfully triggered copy for all @ids passed {success}')
+        return success, errors
+
+    def restore_glacier_phase_three_patch(self, atid_list: List[Union[str, dict]],
+                                          status: str = 'uploaded') -> (List[str], List[str]):
+        """ Patches out lifecycle information for @ids we've transferred back to standard
+
+        :param atid_list: list of @ids or actual file metadata objects to patch info on
+        :param status: top level status to replace for files
+        :return: 2 tuple of success, error list of @ids
+        """
+        success, errors = [], []
```

Comment:
For phase 3 here and phase 4 below, consider double-checking a copied, non-glacier form of the object exists at the location prior to execution.